node/node_test.go

```diff
@@ -57,6 +57,50 @@ func newTestNode(ctx context.Context, mn mocknet.Mocknet, t *testing.T) *node {
 	return nd
 }
 
+type ExchangeOption func(*exchange.Options)
+
+func WithFilecoinAPI(api *filecoin.MockLotusAPI) ExchangeOption {
+	return func(e *exchange.Options) { e.FilecoinAPI = api }
+}
+
+func WithRegions(regions []exchange.Region) ExchangeOption {
+	return func(e *exchange.Options) { e.Regions = regions }
+}
+
+func WithCapacity(newCapacity uint64) ExchangeOption {
+	return func(e *exchange.Options) { e.Capacity = newCapacity }
+}
+
+func (n *node) OverrideExchange(ctx context.Context, t testing.TB, opts ...ExchangeOption) *node {
```

Comment:
I wouldn't add an OverrideExchange method to the node struct just for testing purposes. Can you just have those options on the `newTestNode` function? Seems simpler to me

Reply:
Indeed it's way simpler like that